### DISCUSSION TITLE
x-model initial value for select as the first option's value

### DISCUSSION THREAD

**Discussion by on3ss** [alpinejs/alpine#2737] [1. Help] [answered]
How do I initialize an x-model variable with the first option's value in a select field?

**Comment by SimoTod** [comment, marked as answer]
Alpine is the source of truth so you would generate the component with the right value not the other way around. If not possible, I think something like this would work:
```html
<select x-model="foo" x-init="foo = $el.options[$el.selectedIndex || 0].value">
```

**Reply by on3ss** [reply]
That works. Thanks

**Reply by gharabaghi** [reply]
The above code may not work in some situations. below is working code:
`<select x-model="foo" x-init="$nextTick(() => { foo=$el.value })"">`

**Reply by ekwoka** [reply]
In the latest version of Alpine, you can do `x-model.fill="foo"` to get this behavior.
